kong/dao/db/postgres.lua

```diff
@@ -26,6 +26,27 @@ end
 
 local _M = require("kong.dao.db").new_db("postgres")
 
+local forced_luasocket_phases = {
+  init        = true,
+  init_worker = true,
+}
+
+do
+  local function check_phase(phase)
+    if type(phase) ~= 'string' then
+      local info = debug.getinfo(2)
+      local err = string.format("bad argument #1 to '%s' (%s expected, got %s)",
+                                info.name, 'string', type(phase))
+      error(err, 3)
+    end
+  end
+
+  function _M.force_luasocket(phase, force)
+    check_phase(phase)
+    forced_luasocket_phases[phase] = force
+  end
+end
```

Comment:
As pointed out on another channel: let's keep this feature in this DB module to use the DAO in `init_worker` (pgmoon only automatically falls back on LuaSocket in `init`).